src/common/pico_binary_info/include/pico/binary_info/structure.h

```diff
@@ -21,6 +21,27 @@ extern "C" {
 #define __packed __attribute__((packed))
 #endif
 
+#if !defined(le_uint16_t) || !defined(le_uint32_t) || !defined(le_int32_t)
```

Comment:
seems like this can go in platform.h, no?

Reply:
Yes, but the thing is that `structure.h` does not include `platform.h`.  Which is why it already has its own definition of `__packed` just above.

In the previous version of the patch, I did add an include of `platform.h` here, but then I noticed the comment at the start which says that this file is not supposed to use SDK includes...

Reply:
ah yes; danger of reading in github - yes we should keep that as you have it